applications/smart-distancing/libs/loggers/csv_logger.py

```diff
@@ -0,0 +1,149 @@
+import csv
+import os
+from datetime import date
+
+import numpy as np
+
+
+def prepare_object(object_item, frame_number):
+    """construct a dictionary that is appropriate for csv writer.
+
+    this function transform a dictionary with list values to a dictionary
+    with scalar values. This transformation is necessary for csv writer to avoid
+    writing lists into csv.
+
+    Args:
+        object_item: it is a dictionary that contains an detected object information after postprocessing
+        frame_number: current frame number
+
+    Returns:
+        A transformed version of object_item to a dictionary with only scalar values. It also contains an item
+        for frame number.
+
+    """
+    object_dict = {}
+    object_dict.update({"frame_number": frame_number})
+    for key, value in object_item.items():
+        if isinstance(value, (list, tuple)):
+            for i, item in enumerate(value):
+                if isinstance(item, float):
+                    item = round(item, 4)
+                elif isinstance(item, np.float32):
+                    item = round(float(item), 4)
```

Comment:
I think it does work for both types
item = round(float(item, 4))
And If you want to check the instance type you can write something like this:
if (isinstance(item, float) and isinstance(item, npfloat32)): item = round(float(item, 4))